### ISSUE TITLE
 9.2.4.4 Aussagekräftige Linktexte und title

### ISSUE THREAD

**Issue by mayerhh** [BIK-BITV/BIK-Web-Test#8]
Bei der Prüfung von Links auf andere Dateiformate ist die Verwendung von `title`ausreichend für die Kennzeichnung.
Für Linktexte wie "mehr" gehört der `title` dagegen nicht zu den aufgezählten Möglichkeiten für die Differenzierung.

Das erscheint etwas unlogisch. Es sollte entweder für beide oder keinen der Fälle gelten.

**Comment by Michael-Detmers** [comment]
Ich habe das so verstanden, dass es sich im Download-Fall tatsächlich um Zusatzinformation handelt. Dann wäre `title` ja ausreichend. 

Bei einem "mehr"-Link muss ich ja wissen, wohin die Reise geht, was eine Hauptinformation wäre. Dass diese dann auch "nur" aus dem Kontext gebildet werden kann, ist noch mal eine andere Diskussion.

**Comment by johannesFischer84** [comment]
Ich hatte vor ein paar Jahren angezweifelt, ob das Dateiformat überhaupt angegeben werden muss. Denn aus den WCAG geht das für mich nicht direkt hervor. In unseren Diskussionen wurde dafür argumentiert, dass das Dateiformat ebenso wichtig wie das Inhaltliche für den Nutzer ist, sodass das Dateiformat auch einen Teil des Linkzwecks darstellt. Das habe ich verstanden, denn das ist dann einfach die Frage, wie weit man das Erfolgskriterium auslegt, wie man es interpretiert.

Dennoch würde ich immer noch sagen, dass das Dateiformat eher in erweiterter Form den Linkzweck angibt, weil das in den WCAG nicht klar benannt ist. Der Hauptzweck ist für mich immer noch das Inhaltliche, was bei der aufgerufenen Seite oder der Datei zu finden ist. Deswegen habe ich nichts Schlimmes dabei gefunden, `title` als ausreichend für die Kennzeichnung des Dateiformats anzusehen.

**Comment by sweckenmann** [comment]
> Ich habe das so verstanden, dass es sich im Download-Fall tatsächlich um Zusatzinformation handelt. Dann wäre title ja ausreichend.

Stimmt. Allerdings: Sehende Tastaturnutzer:innen bekommen diese Information nicht über `titel` angezeigt - was ich schon immer etwas problematisch fand.  Der Kontext vermittelt in diesem Fall ja nicht zwingend, was passieren wird. Da es jedoch die Sufficient Technique [H33: Supplementing link text with the title attribute](https://www.w3.org/WAI/WCAG21/Techniques/html/H33.html) gibt, die 2.4.4 Link Purpose (in Context) zugeordnet ist, ist `title` wohl gemäß WCAG ausreichend.

Aufgrund dieser Sufficient Technique, müsste man dann eigentlich auch `title` als eine weitere Möglichkeit (neben z.B. Überschriften im Kontext) gelten lassen, um "mehr"-Links aussagekräftig zu machen.

**Comment by Michael-Detmers** [comment]
Ich glaube niemand ist mit der Implementation für `title` glücklich. Ich wünsche da würde man mal den Standard anpassen, sodass das Attribut auch via Touch brauchbar ist und alle Kriterien von [Eingeblendete Inhalte bedienbar](https://github.com/BIK-BITV/BIK-Web-Test/blob/master/Pr%C3%BCfschritte/de/9.1.4.13%20Eingeblendete%20Inhalte%20bedienbar.adoc) berücksichtigt.

> Aufgrund dieser Sufficient Technique, müsste man dann eigentlich auch `title` als eine weitere Möglichkeit (neben z.B. Überschriften im Kontext) gelten lassen, um "mehr"-Links aussagekräftig zu machen.

Dazu steht ja auf [H33: Supplementing link text with the title attribute](https://www.w3.org/WAI/WCAG21/Techniques/html/H33.html):

> The `title` attribute is used to provide additional information to help clarify or further describe the purpose of a link. If the supplementary information provided through the `title` attribute is something the user should know before following the link, such as a warning, then it should be provided in the link text rather than in the `title` attribute. 

_Help clarify_ oder _further describle the purpose_ impliziert ja, dass der Link grundlegend beschrieben ist und diese Information ergänzt wird. Bei "mehr" kann das nicht der Fall sein, da der Text alleinstehend nichts aussagt.

Also: 
- Alles was vor dem Klick zwingend gewusst werden muss → Linktext
- Zusatzinformation, ohne die der Link weiter verständlich und unterscheidbar ist oder welche die Bedeutung des Linktextes verdeutlicht → `title`

**Comment by sweckenmann** [comment]
> Bei "mehr" kann das nicht der Fall sein, da der Text alleinstehend nichts aussagt.

Naja der Link muss entsprechend [2.4.4 Link Purpose (in Context)](https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=244#link-purpose-in-context) nicht alleinstehend aussagekräftig sein. Das Erfolgskriterium unterscheidet sich von dem AAA-Erfolgkriterium [Link Purpose (Link only)](https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=244#link-purpose-link-only) dadurch, dass der programmatisch ermittelbare Kontext hinzugenommen wird, 2.4.4 sagt:

> The purpose of each link can be determined from the link text alone or from the link text together with its programmatically determined link context, except where the purpose of the link would be ambiguous to users in general.

Andersrum könnte man über die Interpretation von H88 streiten: 

> ...If the supplementary information provided through the title attribute is something the user should know before following the link, such as a warning, then it should be provided in the link text rather than in the title attribute.

Sollte der/die Nutzer:in wissen, dass es sich um ein PDF-Link handelt, befor er/sie dem Link folgt?

Da ich `title` keine gute Technik finde, wäre mir gelgen, wenn wir dies eher weniger als mehr zur Erfüllung eines EK akzeptieren. Kann auch mit der bisherigen Regelung (mehr-Links und PDF-Links unterschiedlich handhaben) leben, aber eine gewisse Unlogik gibt es schon.

**Comment by Michael-Detmers** [comment]
Stimmt, dann ist aber wie zuvor das Kernproblem, ob man `title` als unmittelbaren, zu berücksichtigen Kontext ansieht. Da das Attribut aber nur für Zusatzginweise verwendet werden soll, würde ich bei "`title` reicht nicht aus" bleiben.

Bei den Download-Links ist es wirklich schwierig und ich würde bei Konzeptionen immer dazu raten Dateiname, -Format und -Größe anzugeben (vorausgesetzt die Datei ist barrierefrei - sonst dafür auch noch ein Hinweis). Wenn das Ziel eines Downloadlinks generell aber erfüllt ist (z.B. "Imagefilm herunterladen"), fände ich es gerade noch verschmerzbar, dass die zuvor genannten Daten wenigstens im `title` genannt werden, auch wenn das schlechte Usability ist.

Aus Usability-Sicht finde ich allerdings alle Bedienelemente doof, bei denen ich im Umfeld nach dem Sinn und Zweck suchen muss 🙄

**Comment by johannesFischer84** [comment]
> Stimmt, dann ist aber wie zuvor das Kernproblem, ob man title als unmittelbaren, zu berücksichtigen Kontext ansieht. Da das Attribut aber nur für Zusatzginweise verwendet werden soll, würde ich bei "title reicht nicht aus" bleiben.

Heißt das im Ergebnis der Diskussion, es soll nichts in der Prüfschrittanleitung geändert werden?
Oder heißt es, entsprechend der Eingangsfrage (für beide Fälle oder keinen) darf `title` auch nicht ausreichend für die Kennzeichnung des Dateiformats angesehen werden?
Ich wäre mit beidem zufrieden, wobei ich die Angabe des Dateiformats weiterhin als weite Interpretationsauslegung der WCAG halte (Dateiname, Größe und Barrierefreiheitshinweis als sehr, sehr weite Auslegung der WCAG)
